api_calls.py

```diff
@@ -0,0 +1,540 @@
+import http.client
+import json
+import datetime
+import time
+import csv
+
+api_key = '284s83ypFD8LAEu1Y6WFK5peMLz1KF0Y7jSFHizV'
+month = datetime.datetime.now().month
+month = str(month).zfill(2)
+day = datetime.datetime.now().day
+day = str(day).zfill(2)
+year = str(datetime.datetime.now().year)
+
+## Player class ##
+class PlayerStats:
+    def __init__(self, name, team, points, assists, rebounds, blocks, steals, field_goals_pct, three_points_pct, free_throws_pct):
+        self.name = name
+        self.team = team
+        self.points = points
+        self.assists = assists
+        self.rebounds = rebounds
+        self.blocks = blocks
+        self.steals = steals
+        self.field_goals_pct = field_goals_pct
+        self.three_points_pct = three_points_pct
+        self.free_throws_pct = free_throws_pct
+## End of player class ## 
+        
+
+
+## Team stats class ##
+class TeamStats:
+    def __init__(self,team_name, team_points, team_assists, team_rebounds, team_blocks, team_steals, team_field_goals_pct, team_three_points_pct, team_free_throws_pct):
+        self.team_name = team_name
+        self.team_points = team_points
+        self.team_assists = team_assists
+        self.team_rebounds = team_rebounds
+        self.team_blocks = team_blocks
+        self.team_steals = team_steals
+        self.team_field_goals_pct = team_field_goals_pct
+        self.team_three_points_pct = team_three_points_pct
+        self.team_free_throws_pct = team_free_throws_pct
+## End of team stats class ##
+
+
+
+## Team roster class ##
+class TeamRoster:
+    def __init__(self, full_name, position, jersey_number):
+        self.full_name = full_name
+        self.position = position
+        self.jersey_number = jersey_number
+## End of team roster class ##
+        
+
+
+## Team standings class ##
+class TeamStandings:
+    def __init__(self, team_name, wins, losses, team_id):
+        self.team_name = team_name
+        self.wins = wins
+        self.losses = losses
+        self.team_id = team_id
+## End of team standings class ##
+        
+
+
+
+
+
+
+
+## Facade class for the API calls ##
+class GameFacade:
+    ''' Constructor for the class '''
+    def __init__(self):
+        self.api_key = api_key
+        self.connection = http.client.HTTPSConnection("api.sportradar.us")
+
+
+
+
+
+    ### Download methods ###
+    ''' ONLY USE THESE METHODS WHEN INITIALIZING SYSTEM TO DOWNLOAD 
+        NEEDED INFORMATION FOR REDUCING AMOUNT OF API CALLS SUCH AS 
+        GETTING FULL ROSTER, SCHEDULE, AND TEAM NAMES/TEAM ID
+    '''
+    def download_season_schedule(self)-> None:
+        nba_year = datetime.datetime.now().year - 1
```

Comment:
This shouldn't be an issue for us, but I think this would cause issues in the future - if we were to try and make it work at all times of the year.

I could be wrong, though (and I'm motivated to find _something_ to review, so it's probably fine)

Reply:
Shouldn’t be too bad of a fix, to determine what the fiscal nba year, you’d need to see what year October is as that’s the start of NBA seasons and extract the year there. Hasn’t thought of it in the time.